beacon-chain/rpc/eth/node/handlers_peers.go

```diff
@@ -0,0 +1,286 @@
+	if err != nil {
+		if errors.Is(err, peerdata.ErrPeerUnknown) {
+			http2.HandleError(w, "Peer not found: "+err.Error(), http.StatusNotFound)
+			return
+		}
+		http2.HandleError(w, "Could not obtain ENR: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	serializedEnr, err := p2p.SerializeENR(enr)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain ENR: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	p2pAddress, err := peerStatus.Address(id)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain address: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	state, err := peerStatus.ConnectionState(id)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain connection state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	direction, err := peerStatus.Direction(id)
+	if err != nil {
+		http2.HandleError(w, "Could not obtain direction: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if eth.PeerDirection(direction) == eth.PeerDirection_UNKNOWN {
+		http2.HandleError(w, "Peer not found", http.StatusNotFound)
+		return
+	}
+
+	v1ConnState := migration.V1Alpha1ConnectionStateToV1(eth.ConnectionState(state))
+	v1PeerDirection, err := migration.V1Alpha1PeerDirectionToV1(eth.PeerDirection(direction))
+	if err != nil {
+		http2.HandleError(w, "Could not handle peer direction: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	resp := &GetPeerResponse{
+		Data: &Peer{
+			PeerId:             rawId,
+			Enr:                "enr:" + serializedEnr,
+			LastSeenP2PAddress: p2pAddress.String(),
+			State:              strings.ToLower(v1ConnState.String()),
+			Direction:          strings.ToLower(v1PeerDirection.String()),
+		},
+	}
+	http2.WriteJson(w, resp)
+}
+
+// GetPeers retrieves data about the node's network peers.
+func (s *Server) GetPeers(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "node.GetPeers")
+	defer span.End()
+
+	states := r.URL.Query()["state"]
+	directions := r.URL.Query()["direction"]
+
+	peerStatus := s.PeersFetcher.Peers()
+	emptyStateFilter, emptyDirectionFilter := handleEmptyFilters(states, directions)
+
+	if emptyStateFilter && emptyDirectionFilter {
+		allIds := peerStatus.All()
+		allPeers := make([]*Peer, 0, len(allIds))
+		for _, id := range allIds {
+			p, err := peerInfo(peerStatus, id)
+			if err != nil {
+				http2.HandleError(w, "Could not get peer info: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+			if p == nil {
+				continue
+			}
+			allPeers = append(allPeers, p)
+		}
+		resp := &GetPeersResponse{Data: allPeers}
+		http2.WriteJson(w, resp)
+		return
+	}
+
+	var stateIds []peer.ID
+	if emptyStateFilter {
+		stateIds = peerStatus.All()
+	} else {
+		for _, stateFilter := range states {
+			switch strings.ToUpper(stateFilter) {
+			case stateConnecting:
+				ids := peerStatus.Connecting()
+				stateIds = append(stateIds, ids...)
+				continue
```

Comment:
do you still need continue if it's a switch case?

Reply:
nope, fixed